src/main/java/net/neoforged/neoforge/common/CommonHooks.java

```diff
@@ -1558,4 +1561,20 @@ public static Map<RecipeBookType, Pair<String, String>> buildRecipeBookTypeTagFi
         }
         return vanillaMap;
     }
+
+    public static boolean menuValidity(Player player, Level level, BlockPos blockPos, double distance) {
+        return NeoForge.EVENT_BUS.post(new ValidityEvent.Menu(player, level, blockPos, distance)).getResult();
+    }
+
+    public static boolean menuValidity(Player player, Entity entity, double distance) {
+        return NeoForge.EVENT_BUS.post(new ValidityEvent.EntityMenu(player, entity, distance)).getResult();
+    }
+
+    public static boolean menuValidity(Player player, ContainerEntity entity, double distance) {
+        return NeoForge.EVENT_BUS.post(new ValidityEvent.ContainerEntityMenu(player, entity, distance)).getResult();
+    }
+
+    public static boolean menuValidity(Player player, Level level, BlockPos blockPos, double distance, SignBlockEntity signBlockEntity) {
+        return NeoForge.EVENT_BUS.post(new ValidityEvent.Sign(player, level, blockPos, distance, signBlockEntity)).getResult();
+    }
```

Comment:
I believe these should goes to the `IPlayerExtension` instead, and mark the original methods it replaced as `@Deprecated`.
Otherwise, the original checking method should not be inlined in the event constructor, just leave it in the original call site for not breaking specific mixins from old implementation and multiloader mods.

Reply:
> I believe these should goes to the `IPlayerExtension` instead, and mark the original methods it replaced as `@Deprecated`.

That doesn't work, as there are other callers to those methods. If I replaced those, I could just as well have added an "AdjustPlayerReachDistanceEvent"...